Catch SIGCHLD to avoid a zombie when starting the server.

Cherry-picked from upstream CVS (OpenBSD tmux.c r1.74).
---
 tmux.c |    9 +++++++++
 1 file changed, 9 insertions(+)

--- a/tmux.c
+++ b/tmux.c
@@ -246,6 +246,7 @@
 	struct keylist		*keylist;
 	struct env_data		 envdata;
 	struct msg_command_data	 cmddata;
+	struct sigaction	 sigact;
 	char			*s, *shellcmd, *path, *label, *home, *cause;
 	char			 cwd[MAXPATHLEN], **var;
 	void			*buf;
@@ -545,6 +546,14 @@
 		exit(1);
 	}
 
+	/* Catch SIGCHLD to avoid a zombie when starting the server. */
+	memset(&sigact, 0, sizeof sigact);
+	sigemptyset(&sigact.sa_mask);
+	sigact.sa_handler = SIG_IGN;
+	if (sigaction(SIGCHLD, &sigact, NULL) != 0)
+		fatal("sigaction failed");
+
+	/* Initialise the client socket/start the server. */
 	if ((main_ibuf = client_init(path, cmdflags, flags)) == NULL)
 		exit(1);
 	xfree(path);
